transcom/mymove
# B 21016 tpps received totals int

## [Agility ticket](tbd)

## Summary




### How to test

1. Access the app as an office user
2. Find a payment request needing review, and find one with 2 or more service items.
3. Reject one of the service items so you have an amount greater than zero for the rejected and accepted amounts.
4. Finish reviewing the service items so that the payment request is in the reviewed status.
5. Verify that you see the Accepted and Rejected amounts.
![image](https://github.com/user-attachments/assets/b962b771-9937-4d38-b296-ceba2fadf23e)

6. The next status in the flow is sent to gex. Find your payment request in the DB and set the status to be SENT_TO_GEX and the sent_to_gex_at field to be a valid date.
This field is set when in our code when we send off the 858:
 `sentToGexAt := time.Now() `in `pkg/services/payment_request/payment_request_reviewed_processor.go 
`
![image](https://github.com/user-attachments/assets/a745adf1-f3a9-4826-a2c4-dd1277f96313)
7. Refresh your payment request page and verify that you see the accepted, rejected, and sent to gex amounts with dates matching the reviewed_at date for the accepted/rejected and the date matching the sent_to_gex_at field for the sent to GEX amount.
8. The next status in the flow is received by TPPS. Find your payment request in the DB and set the status to be TPPS_RECEIVED and the received_by_gex_at field to be a valid date.
9. Refresh your payment request page and verify that you see the accepted, rejected, and TPPS received amounts with dates matching the reviewed_at date for the accepted/rejected and the date matching the received_by_gex_at field for the TPPS received amount.
![image](https://github.com/user-attachments/assets/7e606ce7-7878-4404-9ee2-2680857878cd)





## Screenshots
Reviewed
![image](https://github.com/user-attachments/assets/b962b771-9937-4d38-b296-ceba2fadf23e)

Sent to GEX
![image](https://github.com/user-attachments/assets/a745adf1-f3a9-4826-a2c4-dd1277f96313)

TPPS Received
![image](https://github.com/user-attachments/assets/ade9287b-7ac1-48c1-b70d-2e869953678e)

![image](https://github.com/user-attachments/assets/a944f9b4-55d1-4efb-a9c8-13973e558b21)
